src/EditorFeatures/Test2/Diagnostics/DiagnosticServiceTests.vb

```diff
@@ -1940,8 +1940,15 @@ class MyClass
                 ' Get cached project diagnostics.
                 Dim diagnostics = diagnosticService.GetCachedDiagnosticsAsync(workspace, project.Id).WaitAndGetResult(CancellationToken.None)
 
-                Assert.Equal(1, diagnostics.Count())
-                Assert.Equal(HiddenDiagnosticsCompilationAnalyzer.Descriptor.Id, diagnostics.Single().Id)
+                ' different behavior between v1 and v2. in v2. in v2, solution crawler never creates non-local hidden diagnostics.
```

Comment:
@heejaechang One thing we may consider in future to find this out implicitly - if analyzer reporting just hidden diagnostics registers at least one compilation or compilation end action, we run it for project analysis. Any analyzer author who then wants to report non-local hidden diagnostics must do so from these actions.
For now, your approach seems fine.